### PR TITLE
Add example covering the conformance mismatch section

### PR DESCRIPTION
**Pull request by mattmassicotte**
I've added a bunch of examples here to cover the content in Common Errors > Protocol Conformance Isolation Mismatch.

Depends on #23

### CODE REVIEW

File: Sources/Examples/ConformanceMismatches.swift [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	// Swift 5 Warning: main actor-isolated instance method 'applyStyle()' cannot be used to satisfy nonisolated protocol requirement
+	// Swift 6 Error: main actor-isolated instance method 'applyStyle()' cannot be used to satisfy nonisolated protocol requirement
+	func applyStyle() {
+	}
+}
+#endif
+
+// MARK: Globally-Isolated Protocol
+
+/// A conforming type that will infer the protocol's global isolation.
```
Comment (ktoso) [inline Sources/Examples/ConformanceMismatches.swift:18]:
```suggestion
/// A type conforming to the global actor annotated `GloballyIsolatedStyler` protocol,
///  will infer the protocol's global actor isolation.
```

File: Sources/Examples/ConformanceMismatches.swift [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+class GloballyIsolatedWindowStyler: GloballyIsolatedStyler {
+	func applyStyle() {
+	}
+}
+
+/// A conforming type that will infer the protocol's global isolation *only*
```
Comment (ktoso) [inline Sources/Examples/ConformanceMismatches.swift:24]: These feel like they need to explain more; without knowing full context I wouldn't know why from looking at this comment.

```suggestion
/// A type conforming to `PerRequirementIsolatedStyler` which has MainActor isolated protocol requirements,
/// will infer the protocol's requirements isolation for methods witnessing those protocol requirements *only*
```

File: Sources/Examples/ConformanceMismatches.swift [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+/// for the satisfying methods.
+class PerRequirementIsolatedWindowStyler: PerRequirementIsolatedStyler {
+	func applyStyle() {
+		// only this is MainActor-isolated
+	}
```
Comment (ktoso) [inline Sources/Examples/ConformanceMismatches.swift:29]: And then add a method here that is not inferred main actor:

```suggestion
	}
		
		func synchronousMethod() {
 		// this method is non-isolated; it is not witnessing any isolated protocol requirement
 	}
```

Comment (mattmassicotte) [reply Sources/Examples/ConformanceMismatches.swift:31]: yeah this is a particularly useful thing to highlight

**Review by ktoso — APPROVED** [review]
LGTM, maybe next time keep changes in separate commit and then squash at merge time?

It was a bit difficult to figure out what changed after the review round.

Either way, looks good I think

**Comment by mattmassicotte** [comment]
Yes I always try to do that. I don’t know why I did it…

**Pull request merged**